tests/integration_tests/functional/test_initrd.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2019 SUSE LLC
+# SPDX-License-Identifier: Apache-2.0
+"""Tests for initd."""
+
+import pytest
+from host_tools.network import SSHConnection
+from framework.microvm import Serial
+
+INITRD_FILESYSTEM = "rootfs"
+
+
+@pytest.mark.xfail(raises=ConnectionError)
+def test_microvm_initrd_with_ssh(
+        test_microvm_with_initrd,
+        network_config):
+    """Check microvm started with an inird has / mounted as rootfs."""
+    vm = test_microvm_with_initrd
+    vm.spawn()
+    vm.basic_config(
+        add_root_device=False,
+        boot_args="random.trust_cpu=on",
+        use_initrd=True
+    )
+    _tap, _, _ = vm.ssh_network_config(network_config, '1')
+    vm.start()
+
+    # Find out  what is the filesystem of /, and make sure initrd is used
+    conn = SSHConnection(vm.ssh_config)
+    ecode, _, _ = conn.execute_command(
+        f"findmnt / | grep -q -w {INITRD_FILESYSTEM}")
+    assert ecode == 0
+
+
+def test_microvm_initrd_with_serial(
+        test_microvm_with_initrd):
+    """Check microvm started with an inird has / mounted as rootfs."""
+    vm = test_microvm_with_initrd
+    vm.jailer.daemonize = False
+    vm.spawn()
+    vm.memory_events_queue = None
+
+    vm.basic_config(
+        add_root_device=False,
+        boot_args='random.trust_cpu=on console=ttyS0 reboot=k panic=1 pci=off',
+        use_initrd=True
+    )
+
+    vm.start()
+    serial = Serial(vm)
+    serial.open()
+    serial.rx(token='login: ')
```

Comment:
This assert seems to be failing when testing on aarch64. Any reason why the serial would not work with the initrd you provided? Do you happen to know why is it failing? Is there something that we have to change to the resources?

Reply:
That's because the aarch64 kernel does not support initrd loading. You should update it on S3.

Reply:
That's because the aarch64 kernel does not support initrd loading. You should update it on S3.

Reply:
I just checked yesterday with @dhrgit and he says  he uploaded a kernel with support for initrd for aarch64 as well. I also looked at the S3 bucket and there's minimal_with_initrd also for aarch64. Can it be that the serial does not work for another reason? I'll check it tomorrow myself as well. We are attending a meetup these days, sorry for the late reply.

Reply:
Thanks for the info. I assumed the initrd support is missing in kernel because (one month I ago) I did not get any match for `grep initrd /proc/kallsyms` on a aarch64 VM booted with rootfs. If you can check that's great, otherwise I'll try to spend some more time on it.